### PR TITLE
pythonnet: 2.5.2 -> 3.0.1

### PR DESCRIPTION
**Pull request by mdarocha**
###### Description of changes

Updates that also fixes broken build. Also add `clr-loader` package since it's a new dependency.

Helps with https://github.com/NixOS/nixpkgs/issues/234775, since it switches building to .NET Core.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mdarocha** [comment]
@ofborg build python310Packages.pythonnet

File: pkgs/development/python-modules/clr-loader/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    projectFile = [ "netfx_loader/ClrLoader.csproj" "example/example.csproj" ];
+    nugetDeps = ./deps.nix;
+  };
+in
+buildPythonPackage {
+  inherit pname version src;
+
+  format = "pyproject";
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/clr-loader/default.nix:32]:
```suggestion
  postPatch = ''
```

File: pkgs/development/python-modules/pythonnet/default.nix [hunk 2]
```diff
@@ -1,102 +1,78 @@
...
+in
+buildPythonPackage {
+  inherit pname version src;
 
-  preConfigure = ''
-    [ -z "''${dontPlacateNuget-}" ] && placate-nuget.sh
-    [ -z "''${dontPlacatePaket-}" ] && placate-paket.sh
+  format = "pyproject";
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pythonnet/default.nix:34]:
```suggestion
  postPatch = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
otherwise patches might no longer apply

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.clr-loader python310Packages.pythonnet

**Pull request merged**